File: android/motion-gesture-core/src/main/kotlin/io/github/mtatsuto/motiongesture/core/PredictedGestureEvent.kt
package io.github.mtatsuto.motiongesture.core

/** A detector event before trace-specific stream, UUID, and event-sequence fields are assigned. */
data class PredictedGestureEvent(
    val timestampNs: Long,
    val sourceSampleSequence: Long,
    val gesture: Gesture,
)

File: swift/Sources/MotionGestureCore/PredictedGestureEvent.swift
/// A detector event before trace-specific stream, UUID, and event-sequence fields are assigned.
public struct PredictedGestureEvent: Equatable, Sendable {
  public let timestampNs: Int64
  public let sourceSampleSequence: Int64
  public let gesture: Gesture

  public init(timestampNs: Int64, sourceSampleSequence: Int64, gesture: Gesture) {
    self.timestampNs = timestampNs
    self.sourceSampleSequence = sourceSampleSequence
    self.gesture = gesture
  }
}
